contracts/gateway/EIP20Gateway.sol

```diff
@@ -709,44 +693,32 @@ contract EIP20Gateway is GatewayBase {
             "RLP encoded parent nodes must not be zero"
         );
 
-        // Get the redeem intent hash
         bytes32 intentHash = hashRedeemIntent(
             _amount,
-            _beneficiary,
-            _redeemer,
-            _redeemerNonce,
-            _gasPrice,
-            _gasLimit
+            _beneficiary
         );
 
-        // Get the message hash
-        messageHash_ = MessageBus.messageDigest(
-            REDEEM_TYPEHASH,
+        MessageBus.Message memory message = MessageBus.Message(
```

Comment:
Can we do same in `stake`on L:323 
instead of `getMessage`

Reply:
Sure? See https://github.com/OpenSTFoundation/mosaic-contracts/pull/566/files#r243520856